### PR TITLE
Make uuidof() return a static reference

### PR DESCRIPTION
**Pull request by MSxDOS**
These methods currently return GUID by value which causes its unnecessary initialization in the stack instead of referring the value stored in a read-only section e.g. `.rdata`.

https://rust.godbolt.org/z/7bA-yK - a rough example of what's going on.

This is a breaking change but it shouldn't be something drastically hard to follow. The reference may always be dereferenced and then mutably referenced in rare cases where the actual value or a mutable reference is needed since GUID is `Copy`.

### CODE REVIEW

**Comment by retep998** [comment]
This fails on Rust 1.6.0 and is a breaking change anyway, so it will be deferred until winapi 0.4

**Comment by MSxDOS** [comment]
Yeah, the plan for this PR was to be a part of https://github.com/retep998/winapi-rs/issues/885 "pull requests that have breaking changes", should have mentioned that probably.
